# Profitable App Profiles for the App Store and Google Play Markets
## Business Task and Data Source

- The project is an analysis about the profitable App profiles for both the App Store and Google Paly market. 
- We will find out what type of free apps are likely to attract more users, in other words, the most common genres for each market.
- To do this, we'll need to collect and analyze data about mobile apps available on Google Play and the App Store.
- [The data set](https://www.kaggle.com/lava18/google-play-store-apps) from Kaggle containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. It can be downloaded directly from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).
- [The data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) from Kaggle containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. It can be downloaded directly from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

## Opening and Exploring the Data
- In this project, data set from Apple Store is loaded and named `Appstore_data`, and data set for Android apps is named `Googleplay_data`

In [1]:
from csv import reader
##Appstore data
opened_file1=open('../input/app-store-apple-data-set-10k-apps/AppleStore.csv')
Apple_list=reader(opened_file1)
Appstore_data=list(Apple_list)

##Googleplay data
opened_file2=open('../input/google-play-store-apps/googleplaystore.csv')
google_list=reader(opened_file2)
Googleplay_data=list(google_list)

Using the function `explore_data()` to check number of apps in each dataset. 
- Takes in four parameters:
1) `dataset`, which is expected to be a list of lists.
2) `start` and `end`, which are both expected to be integers and represent the starting and the ending indices of a slice from the data set.
3) `rows_and_columns`, which is expected to be a Boolean and has False as a default argument.
- Slices the data set using `dataset[start:end]`.
- Loops through the slice, and for each iteration, prints a row and adds a new line after that row using print('\n').
- Prints the number of rows and columns if `rows_and_columns` is True.

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        print('\n')

In [3]:
explore_data(Appstore_data, 0,1,True)  

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Number of rows: 7198
Number of columns: 17




In [4]:
explore_data(Googleplay_data, 0,1,True) 

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Number of rows: 10842
Number of columns: 13




Appstore has 7197 apps and Google play has 10841 apps.

## Data Cleaning

Before beginning our analysis, we need to make sure the data we analyze is accurate, otherwise the results of our analysis will be wrong. This means that we need to:

- Detect inaccurate data, and correct or remove it.
- Detect duplicate data, and remove the duplicates.

Recall that at our company, we only build apps that are free to download and install, and that are directed toward an English-speaking audience. This means that we'll need to:

- Remove non-English apps like 爱奇艺PPS -《欢乐颂2》电视剧热播.
- Remove apps that aren't free.

### Step 1. Deleting Incorrect Data 
#### Google Play Data
From the discussion https://www.kaggle.com/lava18/google-play-store-apps/discussion we know that there is a missing data.
- Row 10473 in Googleplay_data is not correct. Category is missing.
- delete row 10473

In [5]:
print(Googleplay_data[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [6]:
print(Googleplay_data[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [7]:
len(Googleplay_data[10473])

12

In [8]:
del Googleplay_data[10473]

### Step2. Removing Duplicate Entries 
  
- Created two lists: one for storing the name of duplicate apps, and one for storing the name of unique apps.
- Looped through the data set, and for each iteration:
    1) We saved the app name to a variable named `name`.
    2) If `name` was already in the `unique_apps` list, we appended name to the `duplicate_apps` list.
    3) Else (if name wasn't already in the `unique_apps` list), we appended name to the `unique_apps` list.

#### Google Play Data
- 1181 duplicated entries are found

In [9]:
duplicate_apps = []
unique_apps = []

for app in Googleplay_data[1:]:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicate apps in Google Play is', len(duplicate_apps))
print('Examples of duplicate apps', duplicate_apps[:10])
print('Number of unique apps', len(Googleplay_data)-1-len(duplicate_apps))        

Number of duplicate apps in Google Play is 1181
Examples of duplicate apps ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']
Number of unique apps 9659


#### App Store Data
- doesn't have duplicate entry

In [10]:
duplicate_apps2 = []
unique_apps2 = []

for app in Appstore_data[1:]:
    name = app[0]
    if name in unique_apps2:
        duplicate_apps2.append(name)
    else:
        unique_apps2.append(name)
print('Number of duplicate apps in Apple Store is ', len(duplicate_apps2))
print('Examples of duplicate apps', duplicate_apps2[:10])
print('Number of unique apps', len(Appstore_data)-1-len(duplicate_apps2))

Number of duplicate apps in Apple Store is  0
Examples of duplicate apps []
Number of unique apps 7197


We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

By examining the rows we printed for the *Slack* app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show the data was collected at different times.

In [11]:
for app in Googleplay_data:
    name = app[0]
    if name == 'Slack':
        print(app)

['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51510', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']


- We can use this information to build a criterion for removing the duplicates. The higher the number of reviews, the more recent the data should be. Rather than removing duplicates randomly, we'll only keep the row with the highest number of reviews and remove the other entries for any given app.

- To remove the duplicates, we will:

    - Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app. We use `reviews_max` as the dictionary. The length of dictionary is 9659 which is as expected.
    - Use the information stored in the dictionary and create a new data set, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews).

In [12]:
reviews_max = {}
for app in Googleplay_data[1:]:
    n_reviews = float(app[3])
    name = app[0]
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))  

9659


### Use the dictionary `reviews_max` to remove the duplicate rows:

- Start by creating two empty lists: android_clean (which will store our new cleaned data set) and already_added (which will just store app names).
- Loop through the Google Play data set (make sure you don't include the header row), and for each iteration:
    - Assign the app name to a variable named name.
    - Convert the number of reviews to float, and assign it to a variable named n_reviews.
- If n_reviews is the same as the number of maximum reviews of the app name (the number can be found in the reviews_max dictionary) and name is not already in the list already_added (read the solution notebook to find out why we need this supplementary condition):
- Append the entire row to the android_clean list (which will eventually be a list of list and store our cleaned data set).
- Append the name of the app name to the already_added list — this helps us to keep track of apps that we already added.

In [13]:
android_clean=[]
already_added=[]
for app in Googleplay_data[1:]:
    n_reviews = float(app[3])
    name = app[0]
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(name)
        
print(len(android_clean))

9659


### Step 3. Removing Non-English Apps
- According to the ASCII (American Standard Code for Information Interchange) system, The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127
- Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.
- We can get the corresponding number of each character using the `ord()` built-in function.
- It's very important to minimize the impact of data loss. We will only remove an app if its name has more than 3 non ASCII characters.

#### Google Play Data
- The new dataset after removing duplicate entries from Googleplay data set is named `android_clean`
- I'd like to analyze only the apps that are directed toward an English-speaking audience. However, after exploring the data long enough, I find that both data sets have apps with names that suggest they are not directed toward an English-speaking audience.
- After removing non-English apps from the list, the data set is named `Googleplay_english`

In [14]:
print(android_clean[4412][0])
print(android_clean[7940][0])

中国語 AQリスニング
لعبة تقدر تربح DZ


In [15]:
def char_check(string):
    non_ascii=0
    for char in string:
        if ord(char)>127:
            non_ascii += 1
    if non_ascii >3:
        return False
    else:
        return True

In [16]:
char_check('Instagram')

True

In [17]:
char_check('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [18]:
char_check('Docs To Go™ Free Office Suite')

True

In [19]:
char_check('Instachat 😜')

True

Use the function `char_check()` to filter out non-English apps from both data sets. Loop through each data set. If an app name is identified as English, append the whole row to a separate list.

In [20]:
Googleplay_english = []
for row in android_clean:
    if char_check(row[0]):
        Googleplay_english.append(row)

print(len(Googleplay_english))

9614


### App Store Data
- After removing non-English apps from the list, the data set is named `Appstore_english`

In [21]:
Appstore_english = []
for row in Appstore_data:
    if char_check(row[2]):
        Appstore_english.append(row)

print(len(Appstore_english))

6184


### Step 4. Isolating the Free Apps
- we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps; I'll isolate only the free apps for our analysis.

#### Google Play Data
- According to index number 6 we can isolate the free apps.
- new data set is named `Googleplay_final` and there are 8847 free apps

In [22]:
print(Googleplay_english[0][6])

Free


In [23]:
Googleplay_final = []
for row in Googleplay_english:
    if row[7] == '0':
        Googleplay_final.append(row)
        
print(len(Googleplay_final))

8864


#### Apple Store Data
- Index 5 shows the price. 
- Free apps are isolated by price equals 0
- New dataset is named Appstore_final and the number of free apps is 4056

In [24]:
print(Appstore_data[0])

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [25]:
print(Appstore_english[5][5])

0


In [26]:
Appstore_final=[]
for row in Appstore_english:
    if row[5]=='0':
        Appstore_final.append(row)
        
print(len(Appstore_final))

3222


## Most Common Apps by Genre
Our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, the validation strategy for an app idea is comprised of three steps:

- Build a minimal Android version of the app, and add it to Google Play.
- If the app has a good response from users, we develop it further.
- If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

First, let's begin the analysis by getting a sense of what are the most common genres for each market. I'll build a frequency table for the `prime_genre` column of the App Store data set, and for the `Genres`(index 9) and `Category`(index 1) columns of the Google Play data set.

In [27]:
print(Googleplay_data[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [28]:
print(Googleplay_final[0])

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


There are two functions used to analyze the frequency tables:

- One function to generate frequency tables that show percentages
- Another function we can use to display the percentages in a descending order

In [29]:
def freq_table(dataset,index):
    temp_table = {}
    total = 0
    for row in dataset:
        temp = row[index]
        total += 1
        if temp in temp_table:
            temp_table[temp] += 1
        else:
            temp_table[temp] = 1
            
    table_percentages = {}
    for key in temp_table:
        percent = (temp_table[key] / total) * 100
        table_percentages[key] = percent 
    
    return table_percentages

In [30]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Most Common Genre in App Store

In [31]:
display_table(Appstore_final,-5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that more than half (58.1%) of the apps are in the Games genre, which makes it the most common genre of the App Store.

This is followed by Entertainment (7.9%), then Photo & Video(5.0%), etc...

Most of the free apps designed are dominated by entertainment use such as Games, Photo & Video, Social Networking and Sports. Apps that are designed for practical purposes such as Education, Utilities and Finance are rare and less.

### Most Common Category and Genre in Google Play

In [32]:
display_table(Googleplay_final,1) #Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

The Google Play store has a more significant stand with apps that are designed for practical purposes(family, tools, Business, etc..).

The leading category is family at 18.9%, followed by Game at 9.7% and so on.

In [33]:
display_table(Googleplay_final,9) #Genre

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between the Category column and the Genre column is that the genre column has more categories. As we move forward and looking at the bigger picture, we will only be using the Category column.

Now lets summarize what we know. The App Store is domniated by apps for entertainment purposes and fun, while Google Play store is more diversed and balanced between entertainment apps and practical use apps.

Is this implying a profitable market in the entertainment app develoment? We need to check the number of users.

## Most Popular Apps by Genre on the App Store

- One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the `Installs` column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the `rating_count_tot` app.
- Let's start with calculating the average number of user ratings per app genre on the App Store. To do that, we'll need to:

    - Isolate the apps of each genre.
    - Sum up the user ratings for the apps of that genre.
    - Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

### Number of Apps in Each Genre

In [34]:
def genre_freq_table(dataset,index):
    temp_table = {}
    for row in dataset:
        temp = row[index]
        if temp in temp_table:
            temp_table[temp] += 1
        else:
            temp_table[temp] = 1
    return temp_table

prime_genre_table=genre_freq_table(Appstore_final,-5)

for key, val in sorted(prime_genre_table.items(),key=lambda x:x[1],reverse=True):
    print(key, ':', val)

Games : 1874
Entertainment : 254
Photo & Video : 160
Education : 118
Social Networking : 106
Shopping : 84
Utilities : 81
Sports : 69
Music : 66
Health & Fitness : 65
Productivity : 56
Lifestyle : 51
News : 43
Travel : 40
Finance : 36
Weather : 28
Food & Drink : 26
Reference : 18
Business : 17
Book : 14
Navigation : 6
Medical : 6
Catalogs : 4


### Average Number of User Reviews

In [35]:
count_table={}

for row in Appstore_final:
    rating=float(row[6])
    genre=row[-5]
    if genre in count_table:
        count_table[genre]+=rating
    else:
        count_table[genre]=rating

for genre in count_table:    
    count_table[genre] = count_table[genre]/ prime_genre_table[genre]
    
for g, val in sorted(count_table.items(),key=lambda x:x[1],reverse=True):
    print(g,':',val)

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


The genre "Navigation" has the most reviews but from the frequency table `prime_genre_table` we can see that only 6 apps in this genre.

Navigation is so popular due to the needs of GPS and Car navigation. Waze and Google Maps are dominant.

In [36]:
for row in Appstore_final:
    if row[-5]=='Navigation':
        print(row[2],':',row[6])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


The second most common genre is Reference due to the App 'Bible'.

In [37]:
for row in Appstore_final:
    if row[-5]=='Reference':
        print(row[2],':',row[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


In [38]:
for row in Appstore_final:
    if row[-5]=='Social Networking':
        print(row[2],':',row[6])

Facebook : 2974676
LinkedIn : 71856
Skype for iPhone : 373519
Tumblr : 334293
Match™ - #1 Dating App. : 60659
WhatsApp Messenger : 287589
TextNow - Unlimited Text + Calls : 164963
Grindr - Gay and same sex guys chat, meet and date : 23201
imo video calls and chat : 18841
Ameba : 269
Weibo : 7265
Badoo - Meet New People, Chat, Socialize. : 34428
Kik : 260965
Qzone : 1649
Fake-A-Location Free ™ : 354
Tango - Free Video Call, Voice and Chat : 75412
MeetMe - Chat and Meet New People : 97072
SimSimi : 23530
Viber Messenger – Text & Call : 164249
Find My Family, Friends & iPhone - Life360 Locator : 43877
Weibo HD : 16772
POF - Best Dating App for Conversations : 52642
GroupMe : 28260
Lobi : 36
WeChat : 34584
ooVoo – Free Video Call, Text and Voice : 177501
Pinterest : 1061624
知乎 : 397
Qzone HD : 458
Skype for iPad : 60163
LINE : 11437
QQ : 9109
LOVOO - Dating Chat : 1985
QQ HD : 5058
Messenger : 351466
eHarmony™ Dating App - Meet Singles : 11124
YouNow: Live Stream Video Chat : 12079
Cougar 

In [39]:
for row in Appstore_final:
    if row[-5]=='Music':
        print(row[2],':',row[6])

Pandora - Music & Radio : 1126879
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
Deezer - Listen to your Favorite Music & Playlists : 4677
Sonos Controller : 48905
NRJ Radio : 38
radio.de - Der Radioplayer : 64
Spotify Music : 878563
SoundCloud - Music & Audio : 135744
Sing Karaoke Songs Unlimited with StarMaker : 26227
SoundHound Song Search & Music Player : 82602
Ringtones for iPhone & Ringtone Maker : 25403
Coach Guitar - Lessons & Easy Tabs For Beginners : 2416
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Magic Piano by Smule : 131695
QQ音乐HD : 224
The Singing Machine Mobile Karaoke App : 130
Bandsintown Concerts : 30845
PetitLyrics : 0
edjing Mix:DJ turntable to remix and scratch music : 13580
Smule Sing! : 119316
Amazon Music : 106235
AutoRap by Smule : 18202
My Mixtapez Music : 26286
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
Napster - Top Music 

If we exclude the market of Navigation and Reference, the most popular apps are in the Social Networking and Music category. 

## Most Popular Apps by Genre on Google Play
- Since having the data of the number of installs, it should be easier to abtain the genre popularity. On the other hand, the numbers arent actually precise and mostly left open-ended (1,000,000+, 100,000+, 10,000+, etc..).
- The index number for "Category", "Genres" and "Installs" is 1, -4 and 5
- 2 frequency tables for Category and Genres are created: `cat_freq_table` and `genres_freq_table`

In [40]:
display_table(Googleplay_final,5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In [41]:
cat_freq_table = genre_freq_table(Googleplay_final, 1)
install_table={}
for category in cat_freq_table:
    
    total = 0
    len_category = 0
    for app in Googleplay_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    install_table[category]=avg_n_installs
    #print(category, ':', avg_n_installs)

for g, val in sorted(install_table.items(),key=lambda x:x[1],reverse=True):
    print(g,':',val)

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

The list above tells us that Communication apps has the most installs on average at 38,456,119. 
Followed by Video_Players at 24,727,872 and thirdly Social with 23,253,652 installs.

Each of these three categorys of apps are heavily influenced by big players.

- Communication - WhatsApp, Facebook messenger, Skype, etc...
- Video_Players - Youtube, Google Play Movies & tv, etc...
- Social - Facebook, Instagram, etc...

# Conclusion
- The most popular categories are social networking, music, photograph if excluding the big players